stable/blackbox-exporter/templates/deployment.yaml

```diff
@@ -0,0 +1,108 @@
+{{- $local := dict }}
+{{- if .Values.password -}}
+{{- $_ := set $local "bind" (printf "127.0.0.1:%d" (add1 .Values.service.httpPort)) -}}
+{{- else -}}
+{{ $_ := set $local "bind" (printf ":%0.f" .Values.service.httpPort) -}}
+{{- end -}}
+apiVersion: {{ .Values.apiVersion }}
```

Comment:
Instead of making this configurable we should just go with `apps/v1beta2`.

Reply:
-1. There are 3 versions in active use apps/v1beta2, apps/v1beta1 and apps/v1. I have a production cluster I'm running this chart on that does not support v1beta2+, only v1beta1. I think its 1.9+ that recommends no beta at all or is that 1.10? We should set it to a good default but let it be configurable for a while, until all potential users get up to a more modern version.

Reply:
We officially support N and N-1 (https://github.com/kubernetes/charts#supported-kubernetes-versions), so we only need to use the latest version that works on 1.8. According to the docs, Deployments are apps/v1beta2 there (https://v1-8.docs.kubernetes.io/docs/api-reference/v1.8/).

Reply:
Ok. I'll just keep a fork around for my old cluster. it should be going away in a month or so anyway.

Reply:
Completely agree that forcing the user to set/see the `apiVersion` may not be ideal. Afaik, I only care that the right one is chosen for a given k8s cluster...

Since helm [2.2.0](https://github.com/kubernetes/helm/releases/tag/v2.2.0) it is also possible to check for capabilities. `prometheus-operator` [does this](https://github.com/coreos/prometheus-operator/search?utf8=%E2%9C%93&q=Capabilities.APIVersions&type=) for RBAC. Maybe we should adopt this pattern too.